### ISSUE TITLE
Transformer Evaluation

### ISSUE THREAD

**Issue by dgedon** [antonior92/physionet-12ecg-classification#25]
After learning different transformers, I realized that the combined model in "train.py" does not improve over the model without transformer. 

We can see in [this issue](https://github.com/antonior92/physionet-12ecg-classification/issues/17) that the model without transformer and sigmoid output layer should be able to get to a challenge metric value of about 0.45.

##
So I started to make a series of tests to evaluate and get to the origin of the problem. I have 4 hypotheses of the problem:
1. Evaluating on the small MLP or not finetuning the transformer is not good enough. Hence, try with evaluation on:
- MLP without finetuning, named below `mlp`
- ResNet (without finetuning), named below `resnet`
- ResNet (with finetuning), named below `resnet_finetuning`
2. The new learning method in the transformer does not work properly (despite evidence from a NLP project of mine where I do the same). Hence, do:
- Learn with old transformer learning method, named below `old_full_data`
3. Previously we had a mistake by omitting the dataset "/Training_PTB-XL". So I try to run without this (XL) dataset to see if it causes some problems:
- reduced dataset, named below `reduced_data`
4. In a working version of transformer + resnet, see e.g. [pretrain.py in commit bbc7ddf5c1](https://github.com/antonior92/physionet-12ecg-classification/blob/bbc7ddf5c1181c5efa190303a1dbe3a2032717a9/pretrain.py#L113) the transformer model is much smaller (2 heads, 2 transformer layers, model dimension 50, inner dimension 50). Now I made the transformer as big as it would fit on the GPU, see e.g. current [pretrain.py](https://github.com/antonior92/physionet-12ecg-classification/blob/master/pretrain.py#L115) (4 heads, 3 transformer layers, model dimension 256, inner dimension 768). The big transformer is used for all of the above models. Here, I test a small transformer again with the same dimension as before:
- Small transformer (as successfully used) + reduced dataset, named below `reduced_smallnet`
- Small transformer + full dataset, named below `full_smallnet` (no results yet shown because still running)

##
**Pretrain history**
Note that the `mlp`, `resnet` and `resnet_finetuning` curves are almost on top of each other.

![pretrain_train_loss](https://user-images.githubusercontent.com/43588365/87918409-acaa1200-ca76-11ea-937e-e21d65391a82.png)
![pretrain_valid_loss](https://user-images.githubusercontent.com/43588365/87918423-b3388980-ca76-11ea-9a79-48fbfcaca526.png)

##
**Train history**
![train_loss](https://user-images.githubusercontent.com/43588365/87918857-44a7fb80-ca77-11ea-9d91-cfbb510e0797.png)
![valid_loss](https://user-images.githubusercontent.com/43588365/87918867-48d41900-ca77-11ea-9c7f-393ba3a1c2f3.png)

We observe, that none of the methods actually improves over a challenge metric of 0.0. Specifically:
- no clear conclusion except from the following
- the reduced dataset models do not improve at all. Both remain at ~-0.432. Curves are overlapped

![challenge_metric](https://user-images.githubusercontent.com/43588365/87918873-4b367300-ca77-11ea-9844-1284fb7f63d5.png)
